ui/packages/components/src/Events/EventDetails.tsx

```diff
@@ -0,0 +1,216 @@
+  expandedRowActions: React.ComponentProps<typeof EventsTable>['expandedRowActions'];
+}) {
+  const containerRef = useRef<HTMLDivElement>(null);
+  const leftColumnRef = useRef<HTMLDivElement>(null);
+  const eventInfoRef = useRef<HTMLDivElement>(null);
+  const [leftWidth, setLeftWidth] = useState(70);
+  const [isDragging, setIsDragging] = useState(false);
+
+  const {
+    isPending, // first load, no data
+    error,
+    data: eventDetailsData,
+  } = useQuery({
+    queryKey: ['event-details', { eventName: row.original.name }],
+    queryFn: useCallback(() => {
+      return getEventDetails({ eventName: row.original.name });
+    }, [getEventDetails, row.original.name]),
+  });
+
+  const { error: payloadError, data: eventPayloadData } = useQuery({
+    queryKey: ['event-payload', { eventName: row.original.name }],
+    queryFn: useCallback(() => {
+      return getEventPayload({ eventName: row.original.name });
+    }, [getEventPayload, row.original.name]),
+  });
+
+  const handleMouseDown = useCallback(() => {
+    setIsDragging(true);
+  }, []);
+
+  const handleMouseUp = useCallback(() => {
+    setIsDragging(false);
+  }, []);
+
+  const handleMouseMove = useCallback(
+    (e: MouseEvent) => {
+      if (!isDragging) {
+        return;
+      }
+
+      const container = containerRef.current;
+      if (!container) {
+        return;
+      }
+
+      const containerRect = container.getBoundingClientRect();
+      const newWidth = ((e.clientX - containerRect.left) / containerRect.width) * 100;
+      setLeftWidth(Math.min(Math.max(newWidth, 20), 80));
+    },
+    [isDragging]
+  );
+
+  useEffect(() => {
+    if (isDragging) {
+      document.body.style.userSelect = 'none';
+      window.addEventListener('mousemove', handleMouseMove);
+      window.addEventListener('mouseup', handleMouseUp);
+    }
+    return () => {
+      document.body.style.userSelect = '';
+      window.removeEventListener('mousemove', handleMouseMove);
+      window.removeEventListener('mouseup', handleMouseUp);
+    };
+  }, [isDragging, handleMouseMove, handleMouseUp]);
+
+  if (error || payloadError) {
+    // TODO: error handling
+    console.log(error?.message || payloadError?.message);
+  }
+
+  return (
+    <div ref={containerRef} className="flex flex-row">
+      <div ref={leftColumnRef} className="flex flex-col gap-2" style={{ width: `${leftWidth}%` }}>
+        <div ref={eventInfoRef} className="flex flex-col gap-3">
+          <div className="flex h-8 items-center justify-between gap-1 px-4">
+            <p className="text-muted text-sm">{row.original.name}</p>
+            {expandedRowActions(row.original.name)}
+          </div>
+          <div className="flex flex-row flex-wrap items-center justify-start gap-x-10 gap-y-4 px-4">
+            <ElementWrapper label="Event ID">
+              {isPending ? <SkeletonElement /> : <IDElement>{eventDetailsData?.id}</IDElement>}
+            </ElementWrapper>
```

Comment:
Depending on the actual results of the row data, we could potentially already load this like we do with the row.original.name or we could use the `OptimisticElementWrapper` to pre-render this ID immediately and refresh if anything different.